cores/esp8266/libc_replacements.c

```diff
@@ -84,15 +84,18 @@ int ICACHE_RAM_ATTR _read_r(struct _reent* unused, int file, char *ptr, int len)
 
 int ICACHE_RAM_ATTR _write_r(struct _reent* r, int file, char *ptr, int len) {
     (void) r;
+    int pos = len;
     if (file == STDOUT_FILENO) {
-        while(len--) {
+        while(pos--) {
```

Comment:
Does this address a current issue or ongoing discussion? Maybe a link could be added in the PR description if that's the case.

Reply:
Haven't seen this reported, probably because including iostream header caused compilation error, so no one got as far as using iostreams. Found when writing the test case.